shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -213,14 +248,324 @@ private void copyFile(AlluxioURI srcPath, AlluxioURI dstPath)
     }
   }
 
+  /**
+   * Copies a directory from local to Alluxio filesystem. The destination directory structure
+   * maintained as local directory. This method is used when input path is a directory.
+   *
+   * @param srcDir the source directory in the local filesystem
+   * @param dstPath the {@link AlluxioURI} of the destination
+   * @throws AlluxioException when Alluxio exception occurs
+   * @throws IOException when non-Alluxio exception occurs
+   */
+  private void copyFromLocalDir(File srcDir, AlluxioURI dstPath)
+      throws AlluxioException, IOException {
+    boolean dstExistedBefore = mFileSystem.exists(dstPath);
+    createDstDir(dstPath);
+    List<String> errorMessages = new ArrayList<>();
+    File[] fileList = srcDir.listFiles();
+    if (fileList == null) {
+      String errMsg = String.format("Failed to list files for directory %s", srcDir);
+      errorMessages.add(errMsg);
+      fileList = new File[0];
+    }
+    int misFiles = 0;
+    for (File srcFile : fileList) {
+      AlluxioURI newURI = new AlluxioURI(dstPath, new AlluxioURI(srcFile.getName()));
+      try {
+        copyPath(srcFile, newURI);
+      } catch (AlluxioException | IOException e) {
+        errorMessages.add(e.getMessage());
+        if (!mFileSystem.exists(newURI)) {
+          misFiles++;
+        }
+      }
+    }
+    if (errorMessages.size() != 0) {
+      if (misFiles == fileList.length) {
+        // If the directory doesn't exist and no files were created, then delete the directory
+        if (!dstExistedBefore && mFileSystem.exists(dstPath)) {
+          mFileSystem.delete(dstPath);
+        }
+      }
+      throw new IOException(Joiner.on('\n').join(errorMessages));
+    }
+  }
+
+  /**
+   * Copies a list of files or directories specified by srcFiles from the local filesystem to
+   * dstPath in the Alluxio filesystem space. This method is used when the input path contains
+   * wildcards.
+   *
+   * @param srcFiles a list of source file in the local filesystem
+   * @param dstPath the {@link AlluxioURI} of the destination
+   * @throws AlluxioException when Alluxio exception occurs
+   * @throws IOException when non-Alluxio exception occurs
+   */
+  private void copyFromLocalWildcard(List<File> srcFiles, AlluxioURI dstPath)
+      throws AlluxioException, IOException {
+    boolean dstExistedBefore = mFileSystem.exists(dstPath);
+    createDstDir(dstPath);
+    List<String> errorMessages = new ArrayList<>();
+    int misFiles = 0;
+    for (File srcFile : srcFiles) {
+      AlluxioURI newURI = new AlluxioURI(dstPath, new AlluxioURI(srcFile.getName()));
+      try {
+        copyPath(srcFile, newURI);
+        System.out.println("Copied " + srcFile.getPath() + " to " + dstPath);
```

Comment:
If a user copies `cp file://a b` I think it might be better to say "Copied file://a to b" instead of "Copied a to b". That way the user has more confidence that we did the right thing with the copy. Otherwise they might think that `a` and `b` were both copied inside the alluxio namespace. @maobaolong What do you think?